lib/services/local/presence.go

```diff
@@ -1622,40 +1622,45 @@ func FakePaginate(resources []types.ResourceWithLabels, req proto.ListResourcesR
 		PredicateExpression: req.PredicateExpression,
 	}
 
-	for currIndex, resource := range resources {
-		switch match, err := services.MatchResourceByFilters(resource, filter); {
+	// Iterate and filter every resource, deduplicating while matching.
+	seenResourceMap := make(map[string]struct{})
```

Comment:
Can we have this also be a hashset of pairs of strings? The additional string can just be empty.